kbchat/kvstore.go

```diff
@@ -0,0 +1,152 @@
+package kbchat
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/keybase1"
+)
+
+type GetEntry struct {
+	Result keybase1.KVGetResult `json:"result"`
+	Error  Error                `json:"error,omitempty"`
+}
+
+type PutEntry struct {
+	Result keybase1.KVPutResult `json:"result"`
+	Error  Error                `json:"error,omitempty"`
+}
+
+type DeleteEntry struct {
+	Result keybase1.KVDeleteEntryResult `json:"result"`
+	Error  Error                        `json:"error,omitempty"`
+}
+
+type ListNamespaces struct {
+	Result keybase1.KVListNamespaceResult `json:"result"`
+	Error  Error                          `json:"error,omitempty"`
+}
+
+type ListEntryKeys struct {
+	Result keybase1.KVListEntryResult `json:"result"`
+	Error  Error                      `json:"error,omitempty"`
+}
+
+func (a *API) PutEntry(teamName string, namespace string, entryKey string, entryValue string, revision int) (keybase1.KVPutResult, error) {
+	empty := keybase1.KVPutResult{}
+
+	apiInput := fmt.Sprintf(`{"method": "put", "params": {"options": {"team": "%s", "namespace": "%s", "entryKey": "%s", "entryValue": "%s"}}}`, teamName, namespace, entryKey, entryValue)
+	if revision != 0 {
+		apiInput = fmt.Sprintf(`{"method": "put", "params": {"options": {"team": "%s", "namespace": "%s", "entryKey": "%s", "entryValue": "%s", "revision": %d}}}`, teamName, namespace, entryKey, entryValue, revision)
```

Comment:
i think it would be cleaner to format the `revision` directly into `apiInput` instead of replicating the json in this case